src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,180 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T item, Node<T> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        indexCheck(index);
+        if (index == size) {
+            linkLast(value);
+        } else {
+            linkBefore(value, node(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkPositionIndex(index);
+        MyLinkedList.Node<T> x = node(index);
+        return x.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        indexCheck(index);
+        if (index == size) {
+            throw new IndexOutOfBoundsException();
+        }
+        MyLinkedList.Node<T> x = node(index);
+        if (x != null) {
+            T oldVal = x.item;
+            x.item = value;
+            return oldVal;
+        } else {
+            add(value);
+            return value;
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkPositionIndex(index);
+        MyLinkedList.Node<T> x = node(index);
+        return unlink(x);
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
+            for (MyLinkedList.Node<T> x = first; x != null; x = x.next) {
+                if (x.item == null) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        } else {
+            for (MyLinkedList.Node<T> x = first; x != null; x = x.next) {
+                if (object.equals(x.item)) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        }
```

Comment:
There is no need to split with if/else
```suggestion
            for (MyLinkedList.Node<T> x = first; x != null; x = x.next) {
                if (object == x.item || object != null && object.equals(x.item)) {
                    unlink(x);
                    return true;
                }
        }
```